test/SlowTests/Issues/RavenDB-15693.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System.Threading.Tasks;
+using FastTests;
+using Raven.Client.Documents.Session;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Issues
+{
+    public class RavenDB_15693 : RavenTestBase
+    {
+        private class Doc
+        {
+            public string StrVal1, StrVal2, StrVal3; 
+        }
+        
+        [Fact]
+        public void CanQueryOnComplexBoost()
+        {
+            using var store = GetDocumentStore();
+            using var s = store.OpenSession();
+
+            var q = s.Advanced.DocumentQuery<Doc>()
+                .Search(x => x.StrVal1, "a")
+                .AndAlso()
+                .OpenSubclause()
+                .Search(x => x.StrVal2, "b")
+                .OrElse()
+                .Search(x => x.StrVal3, "search")
+                .CloseSubclause()
+                .Boost(0.2m);
+            var queryBoost = q.ToString();
+            
+            Assert.Equal("from 'Docs' where search(StrVal1, $p0) and boost(search(StrVal2, $p1) or search(StrVal3, $p2), $p3)", queryBoost);
```

Comment:
Query without boost would look like:
```
from 'Docs' where search(StrVal1, $p0) and (search(StrVal2, $p1) or search(StrVal3, $p2))
```

The missing `( )` inside the boost expression will not affect how Lucene generates (and then interprets) query?

Reply:
No, the `()` is there to ensure order of operations, the `boost()` serves the same purpose.